docker/Dockerfile

```diff
@@ -0,0 +1,80 @@
+# When a container is launched from this image with no arguments,
+# the entry point script will launch and initialize a MongoDB instance.
+# It will take a few seconds to be ready for connections, after which
+# radical.pilot will be able to use pymongo to connect.
+# Example:
+#     docker build -t radicalpilot -f Dockerfile ..
+#     docker run --rm --name rp_test -d radicalpilot
+#     # Either use '-d' with 'run' or issue the 'exec' in a separate terminal
+#     # after waiting a few seconds for the DB to be ready to accept connections.
+#     docker exec -ti -u rp rp_test bash -c "cd /radical.pilot && /home/rp/rp-venv/bin/python -m pytest /radical.pilot/tests"
+#     # The examples need the venv to be activated in order to find supporting
+#     # shell scripts on the default PATH. The current working directory also
+#     # needs to be writable.
+#     docker exec -ti -u rp rp_test bash -c "cd && . /home/rp/rp-venv/bin/activate && python /radical.pilot/examples/00*"
+#     # If '-d' was used with 'run', you can just kill the container when done.
+#     docker kill rp_test
+
+FROM mongo:bionic
+# Reference https://github.com/docker-library/mongo/blob/master/4.2/Dockerfile
+
+RUN groupadd -r radical && useradd -r -g radical -m rp
+
+RUN apt-get update && \
+    apt-get install -y \
+        gcc \
+        git \
+        python3-dev \
+        python3-venv \
+        vim && \
+    rm -rf /var/lib/apt/lists/*
+
+USER rp
+
+RUN (cd ~rp && python3 -m venv rp-venv)
+
+RUN (cd ~rp && \
+    rp-venv/bin/pip install --upgrade \
+        pip \
+        setuptools \
+        wheel && \
+    rp-venv/bin/pip install --upgrade \
+        coverage \
+        flake8 \
+        'mock==2.0.0' \
+        netifaces \
+        ntplib \
+        pylint \
+        pymongo \
+        pytest \
+        python-hostlist \
+        setproctitle \
+        )
+
+RUN . ~rp/rp-venv/bin/activate && \
+    pip install --upgrade \
+        'radical.saga>=1.0' \
+        'radical.utils>=1.1'
+
+COPY --chown=rp:radical . /radical.pilot
+
+RUN (. ~rp/rp-venv/bin/activate && \
+    cd /radical.pilot && \
+    ~rp/rp-venv/bin/pip install .)
+
+# radical.saga logs `ModuleNotFountError`s if `redis` is not explicitly installed.
+RUN /home/rp/rp-venv/bin/python -m pip install redis
```

Comment:
The exception appears in the log file, but I don't know whether it is a fatal error. If it is, then either the radical.sage package requirements should be updated, or a more prominent "feature unavailable" sort of error should be contrived.

Reply:
This has recently been addressed via radical-cybertools/radical.saga/pull/783